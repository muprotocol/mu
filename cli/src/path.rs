// TODO: why make this a macro and a wrapper type? The type provides no guarantees anyway
// due to the FromStr implementation, and forces cloning to access the path due to ToString.
#[macro_export]
macro_rules! home_path {
    ($my_struct:ident, $path:literal) => {
        #[derive(Clone, Debug)]
        pub struct $my_struct(String);

        impl Default for $my_struct {
            fn default() -> Self {
                match dirs::home_dir() {
                    None => {
                        println!("$HOME doesn't exist. This probably won't do what you want.");
                        $my_struct(".".to_string())
                    }
                    Some(mut path) => {
                        path.push($path);
                        $my_struct(path.as_path().display().to_string())
                    }
                }
            }
        }

        impl ToString for $my_struct {
            fn to_string(&self) -> String {
                self.0.clone()
            }
        }

        impl FromStr for $my_struct {
            type Err = anyhow::Error;

            fn from_str(s: &str) -> Result<Self, Self::Err> {
                Ok(Self(s.to_string()))
            }
        }
    };
}
