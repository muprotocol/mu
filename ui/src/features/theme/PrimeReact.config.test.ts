import { readFileSync } from "fs";
import {describe, expect, test} from "vitest";
import path from "path";

describe("PrimeReact Config", () => {
    test("the snapshot should not have changed", () => {

        expect(readFileSync(`${__dirname}/PrimeReact.config.ts`)).toMatchSnapshot();
    })
});