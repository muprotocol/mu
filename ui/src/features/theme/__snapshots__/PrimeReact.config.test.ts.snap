// Vitest Snapshot v1

exports[`PrimeReact Config > the snapshot should not have changed 1`] = `
{
  "data": [
    105,
    109,
    112,
    111,
    114,
    116,
    32,
    34,
    112,
    114,
    105,
    109,
    101,
    114,
    101,
    97,
    99,
    116,
    47,
    114,
    101,
    115,
    111,
    117,
    114,
    99,
    101,
    115,
    47,
    116,
    104,
    101,
    109,
    101,
    115,
    47,
    116,
    97,
    105,
    108,
    119,
    105,
    110,
    100,
    45,
    108,
    105,
    103,
    104,
    116,
    47,
    116,
    104,
    101,
    109,
    101,
    46,
    99,
    115,
    115,
    34,
    59,
    10,
    105,
    109,
    112,
    111,
    114,
    116,
    32,
    34,
    112,
    114,
    105,
    109,
    101,
    114,
    101,
    97,
    99,
    116,
    47,
    114,
    101,
    115,
    111,
    117,
    114,
    99,
    101,
    115,
    47,
    112,
    114,
    105,
    109,
    101,
    114,
    101,
    97,
    99,
    116,
    46,
    109,
    105,
    110,
    46,
    99,
    115,
    115,
    34,
    59,
    10,
    105,
    109,
    112,
    111,
    114,
    116,
    32,
    34,
    112,
    114,
    105,
    109,
    101,
    105,
    99,
    111,
    110,
    115,
    47,
    112,
    114,
    105,
    109,
    101,
    105,
    99,
    111,
    110,
    115,
    46,
    99,
    115,
    115,
    34,
    59,
    10,
    105,
    109,
    112,
    111,
    114,
    116,
    32,
    80,
    114,
    105,
    109,
    101,
    82,
    101,
    97,
    99,
    116,
    32,
    102,
    114,
    111,
    109,
    32,
    39,
    112,
    114,
    105,
    109,
    101,
    114,
    101,
    97,
    99,
    116,
    47,
    97,
    112,
    105,
    39,
    59,
    10,
    10,
    80,
    114,
    105,
    109,
    101,
    82,
    101,
    97,
    99,
    116,
    46,
    114,
    105,
    112,
    112,
    108,
    101,
    32,
    61,
    32,
    116,
    114,
    117,
    101,
    59,
    10,
  ],
  "type": "Buffer",
}
`;
